### PR TITLE
Publisher CLI

### PR DESCRIPTION
**Pull request by shavkunov**
Do I need to retry publish message on failure?

### CODE REVIEW

File: src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Publisher.kt [hunk 1]
```diff
@@ -22,7 +22,20 @@ import com.google.cloud.pubsub.v1.Publisher
 import com.google.protobuf.ByteString
 import com.google.pubsub.v1.PubsubMessage
 import com.google.pubsub.v1.TopicName
+import com.xenomachina.argparser.ArgParser
 import org.slf4j.LoggerFactory
+import java.io.File
+
+class PublisherArgs(parser: ArgParser) {
```
Comment (dbarashev) [inline src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Publisher.kt]: CmdLineClient is more appropriate place for these classes, no?

**Comment by dbarashev** [comment]
> Do I need to retry publish message on failure?

No, for command line client we don't need any retries.

File: src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Publisher.kt (hunk 1 above)
Comment (shavkunov) [reply src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Publisher.kt]: Done

File: src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/CmdLineClient.kt [hunk 2]
```diff
@@ -17,14 +17,30 @@ package com.bardsoftware.backend.fleet.rmarkdown
...
 
-fun getRequestData(zipBytes: ByteArray, rootFileName: String, taskId: String): ByteString {
+class PublisherArgs(parser: ArgParser) {
+    val directory by parser.storing(
+            "--dir",
+            help = "directory to be zipped")
+
+    val rootFileName by parser.storing(
+            "-r", "--root-file",
+            help = "name of root rmardownfil ")
```
Comment (dbarashev) [inline src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/CmdLineClient.kt]: This help is alitlbt

File: src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Publisher.kt [hunk 3]
```diff
@@ -65,4 +70,20 @@ class Publisher(private val topicName: String) {
             }
         })
     }
+}
+
+fun main(args: Array<String>) {
```
Comment (dbarashev) [inline src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Publisher.kt]: So, `main` method is also a good fit for CmdLineClient, no?

Comment (shavkunov) [reply src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Publisher.kt]: Yes, indeed
Done.

File: src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/CmdLineClient.kt (hunk 2 above)
Comment (shavkunov) [reply src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/CmdLineClient.kt]: Oops :)
Done

File: src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Publisher.kt [hunk 4]
```diff
@@ -22,7 +22,12 @@ import com.google.cloud.pubsub.v1.Publisher
 import com.google.protobuf.ByteString
 import com.google.pubsub.v1.PubsubMessage
 import com.google.pubsub.v1.TopicName
+import com.xenomachina.argparser.ArgParser
```
Comment (dbarashev) [inline src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Publisher.kt]: Remove unused imports.

**Review by dbarashev — COMMENTED** [review]
LGTM. 
Care to add a task to gradle to start command-line app? 
[See the docs](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.JavaExec.html)

**Review by dbarashev — APPROVED** [review]

File: build.gradle [hunk 5]
```diff
@@ -48,6 +48,14 @@ dependencies {
     //compile 'io.grpc:grpc-protobuf:1.+'
 }
 
+task runApp(type:JavaExec) {
```
Comment (dbarashev) [inline build.gradle]: Pls rename it to `cli` or something like that

**Review by dbarashev — APPROVED** [review]

File: build.gradle (hunk 5 above)
Comment (shavkunov) [reply build.gradle]: Just cli? :)

Comment (dbarashev) [reply build.gradle]: `gradle cli -Dexec.args="-h"` looks pretty neat :)
I don't mind some other name, though, just don't want meaningless "runapp"

Comment (shavkunov) [reply build.gradle]: Done

**Pull request merged**